### ISSUE TITLE
Failed test with v. 0.3.0.1

### ISSUE THREAD

**Issue by wfranzini** [nikita-volkov/hasql-th#10]
Hi,

I'm using hasql-th 0.3.0.1 in one of my projects and today the build failed with the following output.

Is it a known problem and I should migrate to 0.4?

Hope this helps

```
Test suite test: RUNNING...
━━━ Parsing a rendered AST produces the same AST ━━━
  ✓ tableRef passed 10000 tests.
  ✓ aExpr passed 100000 tests.
  ✗ preparableStmt failed at test/Main.hs:28:31
    after 21403 tests, 31 shrinks and 1090205 discards.

       ┏━━ test/Main.hs ━━━
    15 ┃ main = defaultMain [
    16 ┃     checkParallel $ Group "Parsing a rendered AST produces the same AST" $ let
    17 ┃       p _name _amount _gen _parser _renderer =
    18 ┃         (,) _name $ withDiscards (fromIntegral _amount * 200) $ withTests _amount $ property $ do
    19 ┃           ast <- forAll _gen
       ┃           │ SelectPreparableStmt
       ┃           │   (Left
       ┃           │      (SelectNoParens
       ┃           │         Nothing
       ┃           │         (Left
       ┃           │            (TableSimpleSelect
       ┃           │               (SimpleRelationExpr
       ┃           │                  (SimpleQualifiedName (UnquotedIdent "aa")) False)))
       ┃           │         (Just
       ┃           │            (AscDescSortBy
       ┃           │               (ReversableOpAExpr
       ┃           │                  (CExprAExpr
       ┃           │                     (ColumnrefCExpr (Columnref (UnquotedIdent "aa") Nothing)))
       ┃           │                  False
       ┃           │                  (InAExprReversableOp
       ┃           │                     (ExprListInExpr
       ┃           │                        (CExprAExpr
       ┃           │                           (SelectWithParensCExpr
       ┃           │                              (NoParensSelectWithParens
       ┃           │                                 (SelectNoParens
       ┃           │                                    Nothing
       ┃           │                                    (Left
       ┃           │                                       (TableSimpleSelect
       ┃           │                                          (SimpleRelationExpr
       ┃           │                                             (SimpleQualifiedName (UnquotedIdent "aa")) False)))
       ┃           │                                    Nothing
       ┃           │                                    Nothing
       ┃           │                                    Nothing))
       ┃           │                              Nothing) :|
       ┃           │                           []))))
       ┃           │               Nothing
       ┃           │               Nothing :|
       ┃           │               []))
       ┃           │         Nothing
       ┃           │         Nothing))
    20 ┃           let
    21 ┃             sql = Rendering.toText (_renderer ast)
    22 ┃             in do
    23 ┃               footnote ("SQL: " <> Text.unpack sql)
    24 ┃               case Parsing.run _parser sql of
    25 ┃                 Left err -> do
    26 ┃                   footnote err
    27 ┃                   failure
    28 ┃                 Right ast' -> ast === ast'
       ┃                 ^^^^^^^^^^^^^^^^^^^^^^^^^^
       ┃                 │ ━━━ Failed (- lhs) (+ rhs) ━━━
       ┃                 │   SelectPreparableStmt
       ┃                 │     Left
       ┃                 │       SelectNoParens
       ┃                 │         Nothing
       ┃                 │         Left
       ┃                 │           (TableSimpleSelect
       ┃                 │              (SimpleRelationExpr
       ┃                 │                 (SimpleQualifiedName (UnquotedIdent "aa")) False))
       ┃                 │         Just
       ┃                 │ -         AscDescSortBy
       ┃                 │ -           (ReversableOpAExpr
       ┃                 │ -              (CExprAExpr
       ┃                 │ -                 (ColumnrefCExpr (Columnref (UnquotedIdent "aa") Nothing)))
       ┃                 │ -              False
       ┃                 │ -              (InAExprReversableOp
       ┃                 │ -                 (ExprListInExpr
       ┃                 │ -                    (CExprAExpr
       ┃                 │ -                       (SelectWithParensCExpr
       ┃                 │ -                          (NoParensSelectWithParens
       ┃                 │ -                             (SelectNoParens
       ┃                 │ -                                Nothing
       ┃                 │ -                                (Left
       ┃                 │ -                                   (TableSimpleSelect
       ┃                 │ -                                      (SimpleRelationExpr
       ┃                 │ -                                         (SimpleQualifiedName (UnquotedIdent "aa")) False)))
       ┃                 │ -                                Nothing
       ┃                 │ -                                Nothing
       ┃                 │ -                                Nothing))
       ┃                 │ -                          Nothing) :|
       ┃                 │ -                       []))))
       ┃                 │ -           Nothing
       ┃                 │ -           Nothing :|
       ┃                 │ -           []
       ┃                 │ +         AscDescSortBy
       ┃                 │ +           (ReversableOpAExpr
       ┃                 │ +              (CExprAExpr
       ┃                 │ +                 (ColumnrefCExpr (Columnref (UnquotedIdent "aa") Nothing)))
       ┃                 │ +              False
       ┃                 │ +              (InAExprReversableOp
       ┃                 │ +                 (SelectInExpr
       ┃                 │ +                    (WithParensSelectWithParens
       ┃                 │ +                       (NoParensSelectWithParens
       ┃                 │ +                          (SelectNoParens
       ┃                 │ +                             Nothing
       ┃                 │ +                             (Left
       ┃                 │ +                                (TableSimpleSelect
       ┃                 │ +                                   (SimpleRelationExpr
       ┃                 │ +                                      (SimpleQualifiedName (UnquotedIdent "aa")) False)))
       ┃                 │ +                             Nothing
       ┃                 │ +                             Nothing
       ┃                 │ +                             Nothing))))))
       ┃                 │ +           Nothing
       ┃                 │ +           Nothing :|
       ┃                 │ +           []
       ┃                 │         Nothing
       ┃                 │         Nothing
    29 ┃       in [
    30 ┃           p "tableRef" 10000 Gen.tableRef Parsing.tableRef Rendering.tableRef
    31 ┃           ,
    32 ┃           p "aExpr" 100000 Gen.aExpr Parsing.aExpr Rendering.aExpr
    33 ┃           ,
    34 ┃           p "preparableStmt" 35000 Gen.preparableStmt Parsing.preparableStmt Rendering.preparableStmt
    35 ┃         ]
    36 ┃   ]

    SQL: TABLE aa ORDER BY aa IN ((TABLE aa))

    This failure can be reproduced by running:
    > recheck (Size 7) (Seed 12439185917430064721 9856909366553388599) preparableStmt

  ✗ 1 failed, 2 succeeded.
Test suite test: FAIL
Test suite logged to: dist/test/hasql-th-0.3.0.1-test.log
```

**Comment by nikita-volkov** [comment]
Thanks. This will be addressed in future release.

**Comment by nikita-volkov** [comment]
Fixed in 0.4.0.1. Must be noted that the issue was with the test generator, not the library, so it's not necessary to update.
